app/interactors/images/update_crop.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class Images::UpdateCrop
+  include Interactor
+  delegate :params, :user, to: :context
+
+  def initialize(params:, user:)
+    super
+  end
+
+  def call
+    Edition.find_and_lock_current(document: params[:document]) do |edition|
+      image_revision = edition.image_revisions.find_by!(image_id: params[:image_id])
+
+      updater = Versioning::ImageRevisionUpdater.new(image_revision, user)
+      updater.assign(crop_params)
+
+      if updater.changed?
+        update_edition(edition, updater.next_revision)
+        TimelineEntry.create_for_revision(entry_type: :image_updated,
+                                          edition: edition)
+        PreviewService.new(edition).try_create_preview
+      end
+
+      update_context(edition: edition, image_revision: updater.next_revision)
+    end
+  end
+
+private
+
+  def crop_params
+    image_aspect_ratio = Image::HEIGHT.to_f / Image::WIDTH
+    crop_height = params[:crop_width].to_i * image_aspect_ratio
+    # FIXME: this will raise a warning because of unpermitted paramaters
```

Comment:
As per: https://guides.rubyonrails.org/configuring.html#configuring-action-controller these will log a warning in dev and test and do nothing in production (prior to looking this up I assumed it was logged in production too).

We should do something about it as we're not using the feature correctly. Simplest way to get rid of the warning would be `params.slice(:crop_x, :crop_y, :crop_width).permit!`- though I think the most correct way to do it would be to have the params be wrapped.